sygnal/apnspushkin.py

```diff
@@ -328,18 +352,21 @@ def _get_payload_event_id_only(self, n, device):
 
         return payload
 
-    def _get_payload_full(self, n, device, log):
+    def _get_payload_full(
+        self, n: Notification, device: Device, log: NotificationLoggerAdapter
+    ) -> Optional[Dict[str, Any]]:
         """
         Constructs a payload for a notification.
         Args:
             n: The notification to construct a payload for.
-            device (Device): Device information to which the constructed payload
+            device: Device information to which the constructed payload
             will be sent.
             log: A logger.
 
         Returns:
             The APNs payload as nested dicts.
         """
+        assert n.sender is not None
```

Comment:
Unfortunately `sender` and `sender_display_name` are optional in the spec, so the existing code needs updating to handle the case where they're both missing.

https://spec.matrix.org/unstable/push-gateway-api/#post_matrixpushv1notify

Reply:
In the case where we have neither `sender` nor `sender_display_name` would it make sense to just set `from_display` to an empty string, or is there a more graceful way to handle that?

Reply:
There don't seem to be any good options for this case. Whatever we pick will end up spliced into one of these notification strings which may be in any language: https://github.com/vector-im/element-ios/blob/develop/Riot/Assets/en.lproj/Localizable.strings . Anything will do I guess.